### PR TITLE
[ADD]support for password protected redis cluster

### PR DESCRIPTION
**Pull request by etng**
replacement pull request for https://github.com/Grokzen/redis-py-cluster/pull/102

### CODE REVIEW

File: tests/redis-trib.rb [hunk 1]
```diff
@@ -0,0 +1,1402 @@
+#!/usr/bin/env ruby
```
Comment (Grokzen) [inline tests/redis-trib.rb]: @etng Please explain the reason for redis-trib.rb to be included and why not use the built in one that is alreay in redis?

If you have made any modifications to this file, i would really like you to do this instead. First commit, add the file from redis source code and include from what commit hash it was taken. Second commit, include all changes you have done to this file (if any). I can't accept this unless i get that :]

File: rediscluster/nodemanager.py [hunk 2]
```diff
@@ -93,7 +124,7 @@ def initialize(self):
                 startup_nodes_reachable = True
             except ConnectionError:
                 continue
-            except Exception:
+            except Exception as e:
```
Comment (Grokzen) [inline rediscluster/nodemanager.py]: No need for this unless e is used inside the except block

File: rediscluster/nodemanager.py [hunk 3]
```diff
@@ -69,7 +70,37 @@ def random_node(self):
...
+            'ssl_cert_reqs',
+            'ssl_ca_certs',
+            'max_connections',
+        )
+        disabled_keys = (
+            'host',
+            'port',
+            'decode_responses',
+        )
+        connection_kwargs = {k:v for k,v in self.connection_kwargs.iteritems() if k in set(allowed_keys) - set(disabled_keys)}
```
Comment (Grokzen) [inline rediscluster/nodemanager.py]: should be `k: v` and `k, v` for pep8 reasons

File: Makefile [hunk 4]
```diff
@@ -200,7 +357,8 @@ test:
 
 tox:
 	coverage erase
-	tox
+	tox -e py27
```
Comment (Grokzen) [inline Makefile]: I am not sure, but would this not mean that only py27 environment is tested when you run `make tox` ?? There is more environments in there.

**Comment by Grokzen** [comment]
Remember to close the old one if this superseeds it.

File: tests/redis-trib.rb (hunk 1 above)
Comment (etng) [inline tests/redis-trib.rb]: @Grokzen it means thath i should do the followding steps,right?
1. just add the original version of redis-trib.rb from reidis repo,  and commit ,in the comment, add the hash(or version) of the file
2. commit changes I made upon it

**Comment by etng** [comment]
@Grokzen all of my changes are committed now, can you do a code view when you are free?

File: tests/redis-trib.rb (hunk 1 above)
Comment (Grokzen) [inline tests/redis-trib.rb]: Yes that is what i wanted. It makes it alot more clear what you acctually changed to make it work.

**Comment by Grokzen** [comment]
@etng I will review it during the week and it should be merged either before or during the weekend.

**Comment by Grokzen** [comment]
@etng Reviwed this and it all looks good. I pulled the branch locally and i edited some of the commits to do some minor cleanup and i merged it manually so there is no need to merge this PR as is. You still get all the credit and all that even tho the first commit got changed to me O.o dunno why that happened. Still thank you alot for this fix :]

One other question, did you post a PR to add password capability to `redis-trib.rb` so that there is no need to have a patched copy in this repo? If so can you please post that PR so i can track it in another issue to remove it when the support is added.

**Comment by etng** [comment]
@Grokzen my patch for redis-trib.rb is just for create cluster, modify and other operations are not supported now.I will implements these functions and try to send PR the redis project later :)

**Comment by Grokzen** [comment]
Awesome, please try to ping the issue i created about removing the trib file from this repo, so i can track the progress and remove it myself.

**Comment by korvus81** [comment]
Any plan to make another release with this fix in it...?

**Comment by mridhuljp** [comment]
+1

**Comment by otherpirate** [comment]
+1

**Pull request closed without merging**